Tasks/ServiceFabricUpdateManifestsV2/Update-DockerImageSettings.psm1

```diff
@@ -12,18 +12,44 @@ function Update-DockerImageSettings
         $appPackagePathSearchPattern = (Get-VstsInput -Name applicationPackagePath -Require)
         $appPackagePath = Find-VstsFiles -LegacyPattern $appPackagePathSearchPattern -IncludeDirectories
 
-        # Collect the image digest values for the images
+        $imageNamesFilePathSearchPattern = Get-VstsInput -Name imageNamesPath
+        $imageNamesPath = $null
+        if (-not [string]::IsNullOrEmpty($imageNamesFilePathSearchPattern))
+        {
+            $imageNamesPath = Find-VstsFiles -LegacyPattern $imageNamesFilePathSearchPattern
+        }
+
         $imageDigestValues = (Get-Content -LiteralPath $imageDigestsPath).Replace("`r`n", "`n").Split("`n")
+        $imageNames = $null
+        if (($imageNamesPath -ne $null))
+        {
+            $imageNames = (Get-Content -LiteralPath $imageNamesPath).Replace("`r`n", "`n").Split("`n")
+            if ($imageNames.Count -ne $imageDigestValues.Count)
+            {
+                throw (Get-VstsLocString -Key ImageDigestMismatch -ArgumentList @($imageNames.Count, $imageDigestValues.Count))
+            }
+        }
+
+        # Collect the image digest values for the images
         $imageNameToDigestMapping = @{}
-        foreach ($imageDigestValue in $imageDigestValues)
+        for ($ind = 0; $ind -lt $imageDigestValues.Count; $ind++)
         {
-            $slashIndex = $imageDigestValue.IndexOf("/")
-            $hashSeparatorIndex = $imageDigestValue.IndexOf("@")
-            if ($slashIndex -lt 0 -or $hashSeparatorIndex -lt 0)
+            $imageDigestValue = $imageDigestValues[$ind]
+            $imageName = $null
+            if ($imageNames -ne $null)
             {
-                throw (Get-VstsLocString -Key InvalidImageDigestValue -ArgumentList @($imageDigestValue, $imageDigestsPath))
+                $imageName = $imageNames[$ind]
+            }
+            else
```

Comment:
I guess we will never reach else because if $imageNames is $null then line no 27 will throw

Reply:
We will reach this else in the case where no value is provided for $imageNamesPath